packages/graphql/src/index.ts

```diff
@@ -1,24 +1,26 @@
 import { createResource, ResourceReturn } from "solid-js";
 import { access, MaybeAccessor, Modify } from "@solid-primitives/utils";
+import { TypedDocumentNode } from "@graphql-typed-document-node/core";
+import { DocumentNode, print } from "graphql";
 
-export type RequestOptions = Modify<
+export type RequestOptions<V = unknown> = Modify<
   Omit<RequestInit, "body">,
   {
     headers?: RequestHeaders;
-    variables?: QueryVariables;
+    variables?: V;
     fetcher?: typeof fetch;
   }
 >;
 
 export type GraphQLClientQuery = {
-  <T = unknown>(
-    query: string,
-    variables: MaybeAccessor<QueryVariables | boolean> | undefined,
+  <T = unknown, V = unknown>(
+    query: string | DocumentNode | TypedDocumentNode<T, V>,
+    variables: MaybeAccessor<V | boolean> | undefined,
     initialValue: T
   ): ResourceReturn<T, { initialValue: T }>;
-  <T = unknown>(
+  <T = unknown, V = unknown>(
```

Comment:
The `V` generic should extend `{ [key: string]: string | number }`, right? Or at least an object.
```ts
V extends Record<string, string | number> = {}
```

Reply:
The type of filed may be other than string or number. It can be a custom GraphQL scalar type